CODEOWNERS

```diff
@@ -1 +1,7 @@
 *   @jacobtomlinson
+dask_kubernetes/operator/customresources/  @bstadlbauer
+
+# Go operator
+go.mod  @bstadlbauer
+go.sum  @bstadlbauer
+dask_kubernetes/operator/go_client/  @bstadlbauer
```

Comment:
@jacobtomlinson I've added myself as a code-owner to be notified in case of any changes. Happy to revert though